### PR TITLE
Removed redundant Store call in addNOC callback

### LINKED ISSUE #16131: Redundant Store() call in addNOC
#### Problem
We have code that is doing:
```
    err = Server::GetInstance().GetFabricTable().AddNewFabric(gFabricBeingCommissioned, &fabricIndex);
    VerifyOrExit(err == CHIP_NO_ERROR, nocResponse = ConvertToNOCResponseStatus(err));

    err = Server::GetInstance().GetFabricTable().Store(fabricIndex);
```

but AddNewFabric already does a Store() as far as I can tell.

 #### Proposed Solution
Document APIs better, remove redundant Store() call.

### PR DESCRIPTION
**Pull request by harsha-rajendran**
#### Problem
Fixes #16131 
In the AddNOC callback, the same fabric information was being persisted twice, once upon creation in AddNewFabric() and then in a Store() call right after. This PR removes the redundant store call. Also verified that the other AddNewFabric() calls do not have a redundant Store() after.

#### Change overview
Remove redundant Store call

#### Testing
- Verified that the new fabric is not stored twice to the KVS, during commissioning. 
- Sanity-check: Ran OTA test to ensure commissioning and OTA transfer is unaffected

### CODE REVIEW

File: scripts/tests/ota_test.sh [hunk 1]
```diff
@@ -17,6 +17,7 @@ CHIP_TOOL_FOLDER="out"
 
 killall -e "$OTA_PROVIDER_APP" "$OTA_REQUESTOR_APP"
 rm -f "$FIRMWARE_OTA" "$FIRMWARE_BIN" "$OTA_DOWNLOAD_PATH"
+mkdir /tmp/ota/
```
Comment (hrajendran-samsung) [inline scripts/tests/ota_test.sh:20]: Unrelated fixes

**Review by andy31415 — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]
